bears/python/YapfBear.py

```diff
@@ -0,0 +1,135 @@
+        :param split_arguments_when_comma_terminated:
+            Splits before arguments if the argument list is terminated by a
+            comma.
+        :param split_before_bitwise_operator:
+            Set to ``True`` to prefer splitting before ``&``, ``|`` or ``^``
+            rather than after.
+        :param split_before_first_argument:
+            If an argument / parameter list is going to be split, then split
+            before the first argument.
+        :param split_before_logical_operator:
+            Set to ``True`` to prefer splitting before ``and`` or ``or`` rather
+            than after.
+        :param split_before_named_assigns:
+            Splits named assignments into individual lines.
+        :param use_tabs:
+            Use the Tab character for indentation.
+        :param based_on_style:
+            The formatting style to be used as reference.
+        """
+
+        options = """
+        [style]
+        indent_width = {tab_width}
+        column_limit = {max_line_length}
+        allow_multiline_lambdas = {allow_multiline_lambdas}
+        continuation_indent_width = {continuation_tab_width}
+        dedent_closing_brackets = {dedent_closing_brackets}
+        indent_dictionary_value = {indent_dictionary_value}
+        join_multiple_lines = {join_multiple_lines}
+        spaces_around_power_operator = {spaces_around_power_operator}
+        spaces_before_comment = {spaces_before_comment}
+        split_before_bitwise_operator = {split_before_bitwise_operator}
+        split_before_first_argument = {split_before_first_argument}
+        split_before_logical_operator = {split_before_logical_operator}
+        split_before_named_assigns = {split_before_named_assigns}
+        based_on_style = {based_on_style}
+        """
+        options += ('blank_line_before_nested_class_or_def = '
+                    '{blank_line_before_nested_class_or_def}\n'
+                    'space_between_ending_comma_and_closing_bracket = '
+                    '{space_between_ending_comma_and_closing_bracket}\n'
+                    'split_arguments_when_comma_terminated = '
+                    '{split_arguments_when_comma_terminated}\n')
+        options = options.format(**locals())
+
+        with prepare_file(options.splitlines(keepends=True),
+                          None) as (file_, fname):
+            corrected = FormatFile(filename,
+                                   style_config=fname)[0].splitlines(True)
+        diffs = Diff.from_string_arrays(file, corrected).split_diff()
+        for diff in diffs:
+            yield Result(self,
+                         "The code does not comply to the settings provided." +
+                         str(blank_line_before_nested_class_or_def),
+                         affected_code=(diff.range(filename),),
+                         diffs={filename: diff})
+
+    @classmethod
+    def check_prerequisites(cls):
+        return sys.version_info >= (3, 4)
```

Comment:
I think this method can return a message if the prerequisites aren't fulfileld, you should do that so the user knows what's wrong